### ISSUE TITLE
Meanings of the params of mk_forall in OCaml bindings

### ISSUE THREAD

**Issue by Shuumatsu** [Z3Prover/z3#4724]
The document of `mk_forall` in the official documents doesn't mention what are the meanings of the params
```
val mk_forall : context ->
       Sort.sort list ->
       Symbol.symbol list ->
       Expr.expr ->
       int option ->
       Pattern.pattern list ->
       Expr.expr list ->
       Symbol.symbol option ->
       Symbol.symbol option -> quantifier
```

I checked the API documents of Python binding and Haskell binding(unofficial), but these don't take as many params as the OCaml one.
For example, in Haskell: `mkForall: :: Context -> [Pattern] -> [Symbol] -> [Sort] -> AST -> IO AST`  is easy to understand.
Can you explain why the OCaml version needs so many params and what's the meaning of each param?

**Comment by wintersteiger** [comment]
Yeah, documentation in the OCaml API is lacking a bit. You can see the parameter names in [`mk_quantifier`](https://github.com/Z3Prover/z3/blob/1131fedb23a8dd57e932d1c02a570baffa215e6a/src/api/ml/z3.ml#L728). The parameters are the same as in the other APIs and there's a short explanation in the .NET [`MkForall`](http://z3prover.github.io/api/html/class_microsoft_1_1_z3_1_1_context.html#aff0b0c081fb4b14d24550117ef049bfa) for instance (that particular one also has the same order of parameters). 

If you have some spare time, a pull request that adds more documentation (even just copies from the other APIs) would be very welcome!

**Comment by NikolajBjorner** [comment]
I think we want the declaration to use OCaml's features for passing optional arguments.
The default weight should be (Some 1) or just None, the default quantifier_id and Skolem Ids would be None.

  val mk_forall : context -> Sort.sort list -> Symbol.symbol list -> Expr.expr -> ?weight:(int option) -> Pattern.pattern list -> Expr.expr list -> ?quantifier_id:(Symbol.symbol option) -> ?skolem_id:(Symbol.symbol option) -> quantifier
